### DISCUSSION TITLE
How can I use the `<media-theme>` element?

### DISCUSSION THREAD

**Discussion by endymion1818** [muxinc/media-chrome#796] [Q&A] [answered]
I've been following along with creating [themes](https://www.media-chrome.org/docs/en/themes) for media chrome, however it makes sense for my case to instantiate the player in JS, for example:

```javascript
import "media-chrome";

document.querySelector("#app").innerHTML = `
  <div>
      <template id="tiny-theme">
      <media-controller>
        <slot name="media" slot="media"></slot>
        <media-control-bar>
          <media-play-button></media-play-button>
        </media-control-bar>
      </media-controller>
    </template>
    <media-theme template="tiny-theme">
      <video
        slot="media"
        src="https://stream.mux.com/A3VXy02VoUinw01pwyomEO3bHnG4P32xzV7u1j1FSzjNg/high.mp4"
      ></video>
    </media-theme>
  </div>
`;

```

However it doesn't seem to work correctly. It seems to solely render the `<video>` tag without any chrome.

Here's a [CodeSandbox](https://codesandbox.io/p/devbox/kind-cookies-k7sjrg?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cls20y30p00073b6hdsj8v6y6%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cls20y30o00023b6h0ce7sfln%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cls20y30o00043b6hmfbn32lz%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cls20y30p00063b6hx7hydunj%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cls20y30o00023b6h0ce7sfln%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cls20y30o00013b6hctazlvyn%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpnpm-lock.yaml%2522%257D%255D%252C%2522id%2522%253A%2522cls20y30o00023b6h0ce7sfln%2522%252C%2522activeTabId%2522%253A%2522cls20y30o00013b6hctazlvyn%2522%257D%252C%2522cls20y30p00063b6hx7hydunj%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cls20y30p00053b6hsg6t7vaf%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522cls20y30p00063b6hx7hydunj%2522%252C%2522activeTabId%2522%253A%2522cls20y30p00053b6hsg6t7vaf%2522%257D%252C%2522cls20y30o00043b6hmfbn32lz%2522%253A%257B%2522id%2522%253A%2522cls20y30o00043b6hmfbn32lz%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cls20y30o00033b6h8pc6yn9j%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%255D%252C%2522activeTabId%2522%253A%2522cls20y30o00033b6h8pc6yn9j%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) demonstrating the issue

**Comment by luwes** [comment]
Thanks for the interest and question! 

We might have to document this better. The media-theme element requires an explicit import because it adds a few kBs.

This should do it https://codesandbox.io/p/devbox/media-chrome-theme-element-rsnmtt

Hope this helps!

**Reply by endymion1818** [reply]
Thanks for the reply @luwes, unfortunately I can't view that DevBox, could you let me know what I should import? Thanks!

**Reply by endymion1818** [reply]
Actually I found this example to crib from:

https://github.com/muxinc/media-chrome/blob/3e7b0e1bbf666ea79c2decb61ecff8592d56eda7/docs/src/pages/docs/en/themes/share.md?plain=1#L19

**Reply by endymion1818** [reply, marked as answer]
For anyone else searching for this, the solution is to define a template, importing `MediaThemeElement` from media-chrome

```
import { MediaThemeElement } from "media-chrome/dist/media-theme-element";

const template = document.createElement("template");

template.innerHTML = `
    <media-controller>
    <slot name="media" slot="media"></slot>
    <media-control-bar>
        <media-play-button></media-play-button>
    </media-control-bar>
        {{username ?? 'Unknown user'}}
    </media-controller>
`;

class TinyTheme extends MediaThemeElement {
  static template = template;
}

if (!globalThis.customElements.get("tiny-theme")) {
  globalThis.customElements.define("tiny-theme", TinyTheme);
}

export default TinyTheme;
```
Then implement it:
```
import "media-chrome";
import "./tiny-theme";

const username = "bobbytables";

document.querySelector<HTMLDivElement>("#app")!.innerHTML = `
  <tiny-theme username="${username}" >
    <video
      slot="media"
      src="https://stream.mux.com/A3VXy02VoUinw01pwyomEO3bHnG4P32xzV7u1j1FSzjNg/high.mp4"
    ></video>
  </tiny-theme>
`;

```

**Reply by luwes** [reply]
Updated the sandbox with an alternative way with using `<media-theme>` 
https://codesandbox.io/p/devbox/media-chrome-theme-element-rsnmtt
